### PR TITLE
Adding Multiple Activities to Students

### PR DESCRIPTION
**Pull request by jasonvalenzuela**
Story link: https://www.pivotaltracker.com/story/show/166741272

Adds changes to design to allow user to add an activity to multiple students.

### CODE REVIEW

File: src/screens/Students.js [hunk 1]
```diff
@@ -149,25 +163,23 @@ class Students extends Component {
...
-      key={student.id}
-      student={student}
-      onClick={this.handleStudentClick}
-    >
-      <StudentEditButton onClick={this.handleStudentEditClick.bind(null, student)}>
-        <EditIcon/>
-      </StudentEditButton>
-    </StudentListItem>
-  );
+  @action.bound renderStudentListItem(student) {
```
Comment (ryan-peterson-emberex) [inline src/screens/Students.js]: This isn't a "side-effecty" function so it doesn't need the `@action.bound`. Refer to "When to use actions?": https://mobx.js.org/refguide/action.html.

File: src/screens/Students.js [hunk 2]
```diff
@@ -193,80 +205,104 @@ class Students extends Component {
...
               );
-            }}/>
-            
+            }} />
+
+            <Route path={`/${schoolYearId}/students/multiple`} render={props => {
+              const { selectedStudents } = this
+
+              if (selectedStudents.length <= 1) {
+                if (selectedStudents.length === 1) {
```
Comment (ryan-peterson-emberex) [inline src/screens/Students.js]: Do we care if there is one student selected when this "multiple student" functionality is active? I really think we should care about if the list is empty or not, and carry on if there is 1+ student in the list.

File: src/components/EditActivityForm.js [hunk 3]
```diff
@@ -136,15 +136,24 @@ class EditActivityForm extends Component {
       activityToEdit.patch(activity);
       swal('Success', 'Activity saved', 'success');
     } else {
-      this.submitTask = store.createStudentActivity(student, schoolYear, fields);
-      const activity = await this.submitTask;
-      if (onCreateActivity) onCreateActivity(activity);
-      swal('Success', 'Activity created', 'success');
+      if (students) {
+        students.forEach((async (student) => this.createStudentActivity(student, schoolYear, fields)))
```
Comment (ryan-peterson-emberex) [inline src/components/EditActivityForm.js]: I think `await Promise.all(students.map(student => this.createStudentActivity(..fitb..)))` would be a better approach. Alternatively, creating a new endpoint that takes in one or more students and process them on the server would be next level awesome, but this is fine as it is.

File: src/screens/Students.js (hunk 1 above)
Comment (ryan-peterson-emberex) [reply src/screens/Students.js]: Granted, I've seen it used like this. The one benefit is access to `this` without having to think about it, but using an arrow function allows `this` to be bound as you'd want it to.

File: src/components/EditActivityForm.js (hunk 3 above)
Comment (jasonvalenzuela) [reply src/components/EditActivityForm.js]: Definitely! I can make a separate PR in the server repo to add that functionality in.

File: src/screens/Students.js (hunk 2 above)
Comment (jasonvalenzuela) [reply src/screens/Students.js]: From the design, it seems that we want to show the selected student's detail if there is only one student selected: https://projects.invisionapp.com/share/KDT61RTZR5J#/screens/375865786

Comment (ryan-peterson-emberex) [reply src/screens/Students.js]: OIC.

**Comment by jasonvalenzuela** [comment]
This PR on the server will allow for creating multiple activities for students: https://github.com/NTACT/transition-gradebook-server/pull/39

**Pull request merged**